### PR TITLE
Add move mode to bench-tps

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

bench-tps doesn't support move

#### Summary of Changes

Add support.

Fixes #

### CODE REVIEW

File: programs/librapay_api/src/librapay_instruction.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ];
+
+    let invoke_info = InvokeInfo { sender_address: pubkey_to_account_address(from_pubkey), args };
+    let data = bincode::serialize(&invoke_info).unwrap();
+    let ix_data = LoaderInstruction::InvokeMain { data };
+
+    let accounts = vec![AccountMeta::new(*program_id, false), AccountMeta::new(*from_pubkey, true)];
+
+    Instruction::new(solana_move_loader_api::id(), &ix_data, accounts)
+}
```
Comment (garious) [inline programs/librapay_api/src/librapay_instruction.rs]: How about something close to this?

```rust
pub fn transfer(program_id: &Pubkey, from_pubkey: &Pubkey, to_pubkey: &Pubkey, microlibras: u64) -> Instruction {
    let args = vec![
        TransactionArgument::Address(pubkey_to_account_address(to_pubkey)),
        TransactionArgument::U64(microlibras),
    ];

    let ix_data = bincode::serialize(&args).unwrap();

     let accounts = vec![AccountMeta::new(*from_pubkey, true)];

     Instruction::new(program_id, &ix_data, accounts)
}
```

File: programs/move_loader_api/src/lib.rs [hunk 2]
```diff
@@ -59,9 +59,9 @@ pub enum LibraAccountState {
 #[derive(Default, Debug, Serialize, Deserialize)]
 pub struct InvokeInfo {
```
Comment (garious) [inline programs/move_loader_api/src/lib.rs]: Really don't like this struct.  `Vec<TransactionArgument>` should be sufficient.

File: bench-tps/Cargo.toml [hunk 3]
```diff
@@ -23,6 +23,8 @@ solana-metrics = { path = "../metrics", version = "0.17.0" }
 solana-netutil = { path = "../netutil", version = "0.17.0" }
 solana-runtime = { path = "../runtime", version = "0.17.0" }
 solana-sdk = { path = "../sdk", version = "0.17.0" }
+bincode = "1.1.4"
```
Comment (garious) [inline bench-tps/Cargo.toml]: Shouldn't need this

File: bench-tps/src/bench.rs [hunk 4]
```diff
@@ -32,6 +35,8 @@ pub enum BenchTpsError {
     AirdropFailure,
 }
 
+const USE_MOVE: bool = true;
```
Comment (garious) [inline bench-tps/src/bench.rs]: Can you pass this in as a command-line parameter? We're going to want to do native vs Move TPS.

File: programs/librapay_api/src/librapay_transaction.rs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        let program_pubkey = upload_move_program(&mint_keypair, &bank_client);
+        (shared_bank, mint_keypair, program_pubkey)
+    }
+
+    #[test]
+    fn test_payment() {
+        let (bank, alice_keypair, program_id) = create_bank(42);
+        let to = Keypair::new();
+        let tx = payment(&program_id, &alice_keypair, &to.pubkey(), 1, bank.last_blockhash());
+        bank.process_transaction(&tx).unwrap();
```
Comment (garious) [inline programs/librapay_api/src/librapay_transaction.rs]: @jackcmay, check this out. This is a really good test case. We can easily query the bank for `to.pubkey()` just after it and see if that Account's data represents a successful payment. Can you advise on how to write that code? I assume the e2e helper functions?

Comment (jackcmay) [reply programs/librapay_api/src/librapay_transaction.rs]: Looks like this test: https://github.com/solana-labs/solana/blob/8a12ed029cfa38d4a45400916c2463fb82bbec8c/programs/move_loader_api/src/lib.rs#L299

File: bench-tps/src/bench.rs (hunk 4 above)
Comment (sakridge) [reply bench-tps/src/bench.rs]: can just recompile right?

File: programs/move_loader_api/src/lib.rs (hunk 2 above)
Comment (sakridge) [reply programs/move_loader_api/src/lib.rs]: @jackcmay

Comment (jackcmay) [reply programs/move_loader_api/src/lib.rs]: Looking into it

**Comment by garious** [comment]
@sakridge, can you rebase? You should be able to get this through CI now.

**Pull request merged**